lib/egis/client.rb

```diff
@@ -99,16 +105,27 @@ def query_status(query_id)
         query_execution.query_execution_id,
         QUERY_STATUS_MAPPING.fetch(query_status),
         query_execution.status.state_change_reason,
-        parse_output_location(query_execution)
+        parse_output_location(query_execution),
+        output_downloader: output_downloader
       )
     end
 
+    # @!visibility private
+    def output_downloader
+      @output_downloader ||= Egis::OutputDownloader.new(aws_s3_client: aws_s3_client)
+    end
+
+    # @!visibility private
+    def s3_cleaner
+      @s3_cleaner ||= Egis::S3Cleaner.new(aws_s3_client: aws_s3_client)
+    end
```

Comment:
Couldn't we set these in constructor instead? I think that would be more elegant.
If not, could they be private?